src/coreclr/nativeaot/System.Private.StackTraceMetadata/src/Internal/StackTraceMetadata/StackTraceMetadata.cs

```diff
@@ -347,7 +347,8 @@ private unsafe void PopulateRvaToTokenMap(TypeManagerHandle handle, byte* pMap,
                     if ((command & StackTraceDataCommand.UpdateOwningType) != 0)
                     {
                         currentOwningType = Handle.FromIntToken((int)NativePrimitiveDecoder.ReadUInt32(ref pCurrent));
-                        Debug.Assert(currentOwningType.HandleType is HandleType.TypeDefinition or HandleType.TypeReference or HandleType.TypeSpecification);
+                        Debug.Assert((command & StackTraceDataCommand.IsStackTraceHidden) != 0 ||
+                            currentOwningType.HandleType is HandleType.TypeDefinition or HandleType.TypeReference or HandleType.TypeSpecification);
```

Comment:
This one is non-obvious: if TypeDef/TypeSpec/TypeRef handles have their highest bit set, we'd compare them as negative numbers when sorting and StackTraceHidden records (that don't bother filing out owning type) will sort after them. We don't currently see UpdateOwningType command for those since they sort before everything else and owning type == 0 is the initial assumption in the code above (and also in the emitter).